DAMD-128: Add implementation of new PfsReference class

SogoMineo opened the pull request:
The definition of PfsReference is changed so that it will conform to
the revision of datamodel.txt made by DAMD-127 and DAMD-129.

PaulPrice commented on tests/test_FileFormats.py:
```diff
@@ -54,14 +53,6 @@ def testPfsMerged(self):
-        self.assertEqual(d['patch'], '2,2')
-        self.assertEqual(d['catId'], 100)
-        self.assertEqual(d['objId'], 0x02468ace1234abcd)
-
```
Can you modify this test rather than removing it? It would be very simple, and I'm sure you've covered this case in your own tests, but it's good to have it here with other tests of the same kind.

PaulPrice reviewed (changes requested):
We need to make sure this passes the integration test before merging. Currently we can't do that because of a numpy version issue (see below).

SogoMineo commented on python/pfs/datamodel/pfsReference.py:
```diff
@@ -0,0 +1,329 @@
+__all__ = ["PfsReference"]
+
+
+@inheritDocstrings
```
I am deleting this decorator. (I used this decorator because `PfsSimpleSpectrum` is decorated in the same manner.)

SogoMineo commented:
The new `PfsReference` will break a lot of things in `drp_stella` and `pfs_pipe2d`, but there are not yet a code that is utilizing the new `PfsReference`. Now I don't think it a good idea to revise `PfsReference` here, but I feel I had better rename this new `PfsReference` to something like `PfsFluxReference` and keep the old `PfsReference` until the new flux calibration task is made. If I should do so, should I withdraw this PR and open a fresh one, or should I keep using this PR?

SogoMineo commented:
I pushed the redacted version, but it actually breaks `drp_stella` and `pfs_pipe2d`. (See my previous comment).

PaulPrice commented:
`PfsFluxReference` sounds fine. We may as well do it here.

SogoMineo commented:
I revived the old `PfsReference` and renamed the new one to `PfsFluxReference`. I changed the ccommit message accordingly. The integration test now passes. Please review the change.

PaulPrice reviewed (approved):
Awesome!

SogoMineo commented:
I believe that I have now the exact version of astropy that is required.

SogoMineo commented:
Can I merge this PR? The changes are approved here in github, but review is not complete in JIRA.

PaulPrice commented:
Sorry, that's an oversight.
